interfaces/acados_matlab/sim_set_ext_fun.m

```diff
@@ -138,29 +138,37 @@
 		disp(['compiling ', mex_names{ii}])
 		if is_octave()
 	%		mkoctfile -p CFLAGS
-			input_file = fopen('build/cflags_octave.txt', 'r');
+			input_file = fopen(fullfile(build_dir, 'cflags_octave.txt'), 'r');
 			cflags_tmp = fscanf(input_file, '%[^\n]s');
 			fclose(input_file);
 			cflags_tmp = [cflags_tmp, ' -DSET_FIELD=', set_fields{ii}];
 			cflags_tmp = [cflags_tmp, ' -DMEX_FIELD=', mex_fields{ii}];
 			cflags_tmp = [cflags_tmp, ' -DFUN_NAME=', fun_names{ii}];
 			setenv('CFLAGS', cflags_tmp);
-			mex(acados_include, acados_interfaces_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], [acados_mex_folder, 'sim_set_ext_fun_gen.c']);
-			system(['mv sim_set_ext_fun_gen.mex ', mex_names{ii}, '.mex']);
+			mex(acados_include, acados_interfaces_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], fullfile(acados_mex_folder, 'sim_set_ext_fun_gen.c'));
 		else
-			mex(mex_flags, 'CFLAGS=\$CFLAGS -std=c99 -fopenmp', ['-DSET_FIELD=', set_fields{ii}], ['-DMEX_FIELD=', mex_fields{ii}], ['-DFUN_NAME=', fun_names{ii}], acados_include, acados_interfaces_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], [acados_mex_folder, 'sim_set_ext_fun_gen.c']);
-			system(['mv sim_set_ext_fun_gen.mexa64 ', mex_names{ii}, '.mexa64']);
-		end
-
-	end
-
-	if is_octave()
-		system(['mv -f *.o build/']);
-		system(['mv -f *.mex build/']);
-	else
-		system(['mv -f *.mexa64 build/']);
-	end
-
+			mex(mex_flags, 'CFLAGS=$CFLAGS -std=c99 -fopenmp', ['-DSET_FIELD=', set_fields{ii}], ['-DMEX_FIELD=', mex_fields{ii}], ['-DFUN_NAME=', fun_names{ii}], acados_include, acados_interfaces_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], fullfile(acados_mex_folder, 'sim_set_ext_fun_gen.c'));
+    end
+    
+    if is_octave()
+      movefile('sim_set_ext_fun_gen.mex', [mex_names{ii}, '.mex'])
```

Comment:
in `ocp_set_ext_fun.m` this is done with `mexext`: which one is the way to go?

Reply:
i think `mexext` is the way to go, forgot to change it here.

Reply:
resolved